cpp/src/slice2cs/Gen.cpp

```diff
@@ -551,17 +553,34 @@ Slice::CsVisitor::writeConstantValue(const TypePtr& type, const SyntaxTreeBasePt
     else
     {
         BuiltinPtr bp = BuiltinPtr::dynamicCast(type);
-        if(bp && bp->kind() == Builtin::KindString)
+        if(bp)
         {
-            _out << "\"" << toStringLiteral(value, "\a\b\f\n\r\t\v\0", "", UCN, 0) << "\"";
-        }
-        else if(bp && bp->kind() == Builtin::KindLong)
-        {
-            _out << value << "L";
-        }
-        else if(bp && bp->kind() == Builtin::KindFloat)
-        {
-            _out << value << "F";
+            if(bp->kind() == Builtin::KindString)
+            {
+                _out << "\"" << toStringLiteral(value, "\a\b\f\n\r\t\v\0", "", UCN, 0) << "\"";
+            }
+            else if(bp->kind() == Builtin::KindUShort || bp->kind() == Builtin::KindUInt ||
+                    bp->kind() == Builtin::KindVarUInt)
+            {
+                _out << value << "U";
+            }
+            else if(bp->kind() == Builtin::KindLong || bp->kind() == Builtin::KindVarLong)
+            {
+                _out << value << "L";
+            }
+            else if(bp->kind() == Builtin::KindULong || bp->kind() == Builtin::KindVarULong)
+            {
+                _out << value << "UL";
+            }
+            else if(bp->kind() == Builtin::KindFloat)
```

Comment:
What about double?

Reply:
I can add it if you want to be explicit, but in C# a floating-point without a suffix defaults to being a double:
https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/builtin-types/floating-point-numeric-types#real-literals